pkg/proto/proto.go

```diff
@@ -737,26 +729,71 @@ func (a *IpPort) String() string {
 	return NewTCPAddr(a.Addr(), a.Port()).String()
 }
 
+func ipV4Loopback() net.IP {
+	const a, b, c, d = 127, 0, 0, 1
+	return net.IPv4(a, b, c, d)
+}
+
+func filterToIPV4(ips []net.IP) []net.IP {
+	for i := 0; i < len(ips); i++ {
+		ip := ips[i]
+		if ip.IsLoopback() {
+			// after resolving localhost, ip can be in both format, unify it to IPv4
+			ips[i] = ipV4Loopback()
+			continue
+		}
+		if ip.To4() == nil { // for now we support only IPv4
+			ips = append(ips[:i], ips[i+1:]...) // remove non-IPv4 address, assume that count of addresses is small
+		}
+	}
+	return ips
+}
+
+func resolveHostToIPv4(host string) (net.IP, error) {
```

Comment:
Picking a random address isn’t what we need here. The idea of using hostnames for seed peers is that one hostname can point to a list of current seed peers. So, the resulting list should include all the IP addresses resolved from the seed peer’s hostname.

Reply:
Fixed by https://github.com/wavesplatform/gowaves/pull/1551/commits/f03d7020a4a7c4fb7867ac23aba872951f9422ee